src/components/Courses/Courses.jsx

```diff
@@ -0,0 +1,166 @@
+		creationDate: '10/11/2020',
+		duration: 210,
+		authors: [
+			'df32994e-b23d-497c-9e4d-84e4dc02882f',
+			'095a1817-d45b-4ed7-9cf7-b2417bcbf748',
+		],
+	},
+];
+
+const mockedAuthorsList = [
+	{ id: '27cc3006-e93a-4748-8ca8-73d06aa93b6d', name: 'Vasiliy Dobkin' },
+	{ id: 'f762978b-61eb-4096-812b-ebde22838167', name: 'Nicolas Kim' },
+	{ id: 'df32994e-b23d-497c-9e4d-84e4dc02882f', name: 'Anna Sidorenko' },
+	{ id: '095a1817-d45b-4ed7-9cf7-b2417bcbf748', name: 'Valentina Larina' },
+];
+
+/**
+ * Get Authors Names by IDs and join them in one string
+ *
+ * @param {string[]} authorIDs
+ * @returns {string}
+ */
+let getAuthorsNames = (authorIDs, authorsList) => {
+	let names = [];
+	authorIDs.forEach((id) => {
+		let foundAuthor = authorsList.find((e) => e.id === id);
+		names.push(foundAuthor.name);
+	});
+	return names.join(', ');
+};
+
+/**
+ * Search courses by Title or ID
+ *
+ * @returns Object[]
+ */
+let getSearchedCourses = (coursesList) => {
+	// Get Search input value
+	let inputData = document.getElementById('search').value;
+	inputData = inputData.toLowerCase();
+
+	// The array with search result
+	let searchResult = [];
+
+	// Looking for courses by Title or ID
+	coursesList.forEach((course) => {
+		let courseTitle = course.title.toLowerCase();
+		let courseId = course.id.toLowerCase();
+		if (courseTitle.includes(inputData) || courseId.includes(inputData)) {
+			// If found, add to result array
+			searchResult.push(course);
+		}
+	});
+
+	return searchResult;
+};
+
+/**
+ * Courses React component
+ */
+function Courses(props) {
+	const [coursesList, setCoursesList] = useState(
+		props.coursesList ? props.coursesList : mockedCoursesList
+	);
+	const [authorsList, setAuthorsList] = useState(
+		props.authorsList ? props.authorsList : mockedAuthorsList
+	);
+	const [coursesHidden, setCoursesHidden] = useState(props.hidden);
+	const [createCourseFormDisplay, setCreateCourseFormDisplay] = useState(false);
+	const [saveCoursesList, setSaveCoursesList] = useState(coursesList);
+
+	/**
+	 * Find courses by Search input value
+	 */
+	let findCourses = () => {
+		setCoursesList(getSearchedCourses(saveCoursesList));
+	};
+
+	/**
+	 * Display all Courses if Search input is clear
+	 */
+	let clerFilter = () => {
+		// Get Search input value
+		let inputData = document.getElementById('search').value;
+
+		if (!inputData) {
+			setCoursesList(saveCoursesList);
+		}
+	};
+
+	/**
+	 * Render the Create Course Form
+	 */
+	let renderCreateCourseForm = () => {
+		setCoursesHidden(true);
+		setCreateCourseFormDisplay(true);
+	};
+
+	if (!coursesHidden) {
+		return (
+			<div className='courses-wrapper'>
+				<div className='row mt-4'>
+					<div className='col-lg-7 mb-4 mb-lg-0'>
+						<SearchBar actionButton={findCourses} actionInput={clerFilter} />
```

Comment:
Prefer onClick, onChange, onSubmit naming, that's a most used convention

Reply:
fixed